src/contexts/RefinebioContext.js

```diff
@@ -12,8 +17,18 @@ export const RefinebioContextProvider = ({ children }) => {
     []
   )
   const [token, setToken] = useLocalStorage('token', null)
+  const pageRendered = usePageRendered()
   const [downloadOptions, setDownloadOptions] = useState({})
 
+  // NOTE: migration support is removed 12 months after the site swap
+  useEffect(() => {
+    if (pageRendered) {
+      const oldKey = 'dataSetId'
+      setDatasetId(getOldLocalStorageKey(oldKey) || datasetId)
```

Comment:
> <table>
>  <thead>
>    <tr>
>    <th>Old Key</th>
>    <th>New Key</th>
>    <th>Default Value</th>
>    <th>Description</th>
>  </tr>
>  </thead>
>  <tbody>
>   <tr>
>    <td><code>dataSetId</code></td>
>    <td><code>datasetId</code></td>
>    <td><code>null</code></td>
>    <td>A dataset ID which is a UUID string.</td>
>   </tr>
>  </tbody>
> </table>

Thank you for the suggestion, David. That's a good point, but perhaps we can leave this implementation as it is without performing that check 💡 

Here is why:
- Users would not have `datasetId` in their localStorage if they had been using the previous version of refine.bio, which uses `dataSetId`. 

- Once users re-visit refine.bio after the source code swap, the old key `dataSetId` will be no longer used to update the state of datasets. Instead the application will verify if the old key is still present in their localStorage:
    - If so, then it will create a new key`datasetId` in the localStorage, assign the value of the old key to it, and delete that old key from the localStorage. 
    - If not, then it will set the new key `datasetId` value to `null` until the user creates one

In terms of the possibility of overwriting a dataset's state:

- Each browser can store only one dataset ID (a unique key) in its localStorage, so even if multiple tabs are opened, the dataset ID value would remain the same across all tabs.
  - e.g., Since an old key value will be assigned to a new key, the values of `dataSetId`, and `datasetId` will be identical across all tabs. 
 
- Every change users make to their dataset would be immediately saved to the BE database via API requests. The last dataset action (add, remove) users have taken on their dataset will be their most recent dataset state and it will be persisted across tabs.
  - e.g., There are 3 tabs (A, B, C) opened. If a user change something in tab A, then that will be saved to BE and will be the latest dataset state in all tabs - this is applicable for both `dataSetId` and `datasetId` (also it's probably unlikely that a user will instantly go from one tab to another and perform different dataset actions back to back).

- On every page load (for pages that display users' dataset details), to keep users' dataset in sync with the BE, the application makes an API request to fetch the most recent dataset state associated with the dataset ID stored in their localStorage. 
 
For these reasons, I think it's safe to keep the implementation as is? Please let me know your thought on this 💭 , thank you David.

Reply:
I think you are probably right, but in general if we don't need to set a value we shouldn't. Please add the check for the new value before setting it and we can get this merged.

Reply:
I included the check for the value of the old key `dataSetId`. If it exists, assign it to the new key `datasetId` and delete that old key from localStorage (b594571c62ecadc0db06f297b83dc8fd5c334d26), otherwise no allocation and deletion is performed.